### PR TITLE
Add concepts/containers/images.md in pt language

### PR DESCRIPTION
**Pull request by HugoPfeffer**
Add translation of "Imagens de Contêineres"(Container Images)

Add Concepts/containers/images.md

Changes to be committed:
new file: content/pt/docs/concepts/containers/images.md

Ref: https://docs.google.com/spreadsheets/d/1HiaVFJ7nRZo1MPNNYyq1u-L_N9xrJvzuUzkZ3GuJGNU/edit#gid=325775340

Preview: https://deploy-preview-26789--kubernetes-io-master-staging.netlify.app/pt/docs/concepts/containers/images/

/language pt

### CODE REVIEW

**Comment by edsoncelio** [comment]
Hey @HugoPfeffer, thanks for that translation!
Just a comment,  I think to match with the en version, the directory content/pt/docs/concepts/**container**/ need to be content/pt/docs/concepts/**containers**/

**Comment by HugoPfeffer** [comment]
Hey @edsoncelio, thanks for pointing it out!
Do you know how can I fix this? This is my first PR on a big repo, I'm still getting used to all of the CI/CD stuff :)

**Comment by devlware** [comment]
Hi @HugoPfeffer, thanks for helping the project. I'll take a look at your PR later today. :)

File: content/pt/docs/concepts/containers/images.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+## Nomes das imagens
+
+As imagens de contêiner geralmente recebem um nome como `pause`, `exemplo/meuconteiner`, ou `kube-apiserver`.
+As imagens também podem incluir um hostname de algum registro; por exemplo: `exemplo.registro.ficticio/nomeimagem`,
+e um possível número de porta; por exemplo: `exemplo.registro.ficticio:10443/nomeimagem`.
+
+Se você não especificar um hostname de registro, o Kubernetes presumirá que você se refere ao registro público do Docker.
+
+Após a parte do nome da imagem, você pode adicionar uma _tag_ (como também usar com comandos como
+como `docker` e` podman`).
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:30]: In this line I believe there is an extra 'como'.

File: content/pt/docs/concepts/containers/images.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+As tags permitem identificar diferentes versões da mesma série de imagens.
+
+Tags de imagem consistem em letras maiúsculas e minúsculas, dígitos, sublinhados (`_`),
+pontos (`.`) e travessões (` -`).
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:34]: Nitpick, there is an extra space inside the ` ` for -, it should be (`-`).

File: content/pt/docs/concepts/containers/images.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+{{< caution >}}
+Você deve evitar usar a tag `latest` quando estiver realizando o deploy de contêineres em produção,
+pois é mais difícil rastrear qual versão da imagem está sendo executada, além de tornar mais difícil o processo de reversão para uma versão funcional.
+
+Em vez disso, especifique uma tag significativa, como `v1.42.0`.
+{{< /caution >}}
+
+## Atualizando imagens
+
+A política de pull padrão é `IfNotPresent` que faz com que o
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:48]: Suggestion, there is another option for this.
```suggestion
A política padrão de pull é `IfNotPresent` a qual faz com que o
```

File: content/pt/docs/concepts/containers/images.md [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+- defina a `imagePullPolicy` do contêiner para` Always`.
+- omita `imagePullPolicy` e use`: latest` como a tag para a imagem a ser usada.
+- omita o `imagePullPolicy` e a tag da imagem a ser usada.
+- habilite o [AlwaysPullImages](/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages) controlador de admissão.
+
+Quando `imagePullPolicy` é definido sem um valor específico, ele também é definido como` Always`.
+
+## Multiarquitetura de imagens com índice de imagens
+
+Além de fornecer o binário das imagens, um registro de contêiner também pode servir um [índice de imagem do contêiner](https://github.com/opencontainers/image-spec/blob/master/image-index.md). Um índice de imagem pode apontar para multiplos [manifestos da imagem](https://github.com/opencontainers/image-spec/blob/master/manifest.md) para versões específicas de arquitetura de um contêiner. A ideia é que você possa ter um nome para uma imagem (por exemplo: `pause`, `exemple/meuconteiner`, `kube-apiserver`) e permitir que diferentes sistemas busquem o binário da imagem correta para a arquitetura de máquina que estão usando.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:62]:
```suggestion
Além de fornecer o binário das imagens, um registro de contêiner também pode servir um [índice de imagem do contêiner](https://github.com/opencontainers/image-spec/blob/master/image-index.md). Um índice de imagem pode apontar para múltiplos [manifestos da imagem](https://github.com/opencontainers/image-spec/blob/master/manifest.md) para versões específicas de arquitetura de um contêiner. A ideia é que você possa ter um nome para uma imagem (por exemplo: `pause`, `exemple/meuconteiner`, `kube-apiserver`) e permitir que diferentes sistemas busquem o binário da imagem correta para a arquitetura de máquina que estão usando.
```

File: content/pt/docs/concepts/containers/images.md [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+O próprio Kubernetes normalmente nomeia as imagens de contêiner com o sufixo `- $ (ARCH)`. Para retrocompatibilidade, gere as imagens mais antigas com sufixos. A ideia é gerar a imagem `pause` que tem o manifesto para todas as arquiteturas e `pause-amd64` que é retrocompatível com as configurações anteriores ou arquivos YAML que podem ter codificado as imagens com sufixos.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:64]:
```suggestion
O próprio Kubernetes normalmente nomeia as imagens de contêiner com o sufixo `-$(ARCH)`. Para retrocompatibilidade, gere as imagens mais antigas com sufixos. A ideia é gerar a imagem `pause` que tem o manifesto para todas as arquiteturas e `pause-amd64` que é retrocompatível com as configurações anteriores ou arquivos YAML que podem ter codificado as imagens com sufixos.
```

File: content/pt/docs/concepts/containers/images.md [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+   - Especificando ImagePullSecrets em um Pod
+     - apenas pods que fornecem chaves próprias podem acessar o registro privado
+   - Extensões locais ou específicas do fornecedor
+     - se estiver usando uma configuração de nó personalizado, você (ou seu provedor de nuvem) pode implementar seu mecanismo para autenticar o nó ao registro do contêiner.
+
+Essas opções são explicadas com mais detalhes abaixo.
+
+### Configurando nós para autenticação em um registro privado
+
+Se você executar o Docker em seus nós, poderá configurar o contêiner do Docker
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:85]:
```suggestion
Se você executar o Docker em seus nós, poderá configurar o contêiner runtime do Docker
```

File: content/pt/docs/concepts/containers/images.md [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+runtime para autenticação em um registro de contêiner privado.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:86]:
```suggestion
para autenticação em um registro de contêiner privado.
```

File: content/pt/docs/concepts/containers/images.md [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+  1. Visualize `$HOME/.docker/config.json` em um editor para garantir que contém apenas as credenciais que você deseja usar.
+  1. Obtenha uma lista de seus nós; por exemplo:
+      - se você quiser os nomes: `nodes=$( kubectl get nodes -o jsonpath='{range.items[*].metadata}{.name} {end}' )`
+      - se você deseja obter os endereços IP: `nodes=$( kubectl get nodes -o jsonpath='{range .items[*].status.addresses[?(@.type=="ExternalIP")]}{.address} {end}' )`
+  1. Copie seu `.docker/config.json` local para uma das listas de caminhos de busca acima.
+      - por exemplo, para testar isso: `for n in $nodes; do scp ~/.docker/config.json root@"$n":/var/lib/kubelet/config.json; done`
+
+{{< note >}}
+Para clusters de produção, use uma ferramenta de gerenciamento de configuração para que você possa aplicar esta
+configuração para todos os nós onde você precisa.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:123]:
```suggestion
configuração em todos os nós que você precisar.
```

File: content/pt/docs/concepts/containers/images.md [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+Em caso de falha, a saída é semelhante a:
+```
+  Fri, 26 Jun 2015 15:36:13 -0700    Fri, 26 Jun 2015 15:39:13 -0700    19    {kubelet node-i2hq}    spec.containers{uses-private-image}    failed        Failed to pull image "user/privaterepo:v1": Error: image user/privaterepo:v1 not found
+```
+
+
+Você deve garantir que todos os nós no cluster tenham o mesmo `.docker/config.json`. Caso contrário, os pods serão executados com sucesso em alguns nós e falharão em outros. Por exemplo, se você usar o escalonamento automático de nós, cada modelo de instância precisa incluir o `.docker/config.json` ou montar um drive que o contenha.
+
+Todos os pods terão premissão de leitura às imagens em qualquer registro privado, uma vez privado
+as chaves de registro são adicionadas ao `.docker/config.json`.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:169]:
```suggestion
Todos os pods terão premissão de leitura às imagens em qualquer registro privado, uma que as chaves privadas do registro são adicionadas ao `.docker/config.json`.
```

File: content/pt/docs/concepts/containers/images.md [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+
+Isso precisa ser feito para cada pod que está usando um registro privado.
+
+No entanto, a configuração deste campo pode ser automatizada definindo o imagePullSecrets
+em um recurso de [ServiceAccount](/docs/tasks/configure-pod-container/configure-service-account/).
+
+Verifique [Adicionar ImagePullSecrets a uma conta de serviço](/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account) para obter instruções detalhadas.
+
+Você pode usar isso em conjunto com um `.docker / config.json` por nó. As credenciais
+será mesclado.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:255]:
```suggestion
serão mescladas.
```

File: content/pt/docs/concepts/containers/images.md [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+    - Use um [registro Docker](https://docs.docker.com/registry/) privado hospedado.
+      - Pode ser hospedado no [Docker Hub](https://hub.docker.com/signup) ou em outro lugar.
+      - Configure manualmente .docker/config.json em cada nó conforme descrito acima.
+    - Ou execute um registro privado interno atrás de seu firewall com permissão de leitura.
+      - Nenhuma configuração do Kubernetes é necessária.
+    - Use um serviço de registro de imagem de contêiner que controla o acesso à imagem
+      - Funcionará melhor com o escalonamento automático do cluster do que com a configuração manual de nós.
+    - Ou, em um cluster onde alterar a configuração do nó é inconveniente, use `imagePullSecrets`.
+1. Cluster com imagens proprietárias, algumas das quais requerem controle de acesso mais rígido.
+    - Certifique-se de [controlador de admissão AlwaysPullImages] (/ docs / reference / access-authn-authz / admission-controllers / # alwayspullimages) está ativo. Caso contrário, todos os pods têm potencialmente acesso a todas as imagens.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:277]:
```suggestion
    - Certifique-se de que o [controlador de admissão AlwaysPullImages] (/ docs / reference / access-authn-authz / admission-controllers / # alwayspullimages) está ativo. Caso contrário, todos os pods têm potencialmente acesso a todas as imagens.
```

File: content/pt/docs/concepts/containers/images.md [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+    - Mova dados confidenciais para um recurso "secreto", em vez de empacotá-los em uma imagem.
+1. Um cluster multilocatário em que cada locatário precisa de seu próprio registro privado.
+    - Certifique-se de [controlador de admissão AlwaysPullImages](/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages) está ativo. Caso contrário, todos os Pods de todos os locatários terão potencialmente acesso a todas as imagens.
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:280]:
```suggestion
    - Certifique-se de que o [controlador de admissão AlwaysPullImages](/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages) está ativo. Caso contrário, todos os Pods de todos os locatários terão potencialmente acesso a todas as imagens.
```

File: content/pt/docs/concepts/containers/images.md [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+    - Gere credenciais de registro para cada locatário, coloque em segredo e preencha o segredo para cada namespace de locatário.
+    - O locatário adiciona esse segredo a imagePullSecrets de cada namespace.
+
+
+Se precisar de acesso a vários registros, você pode criar um segredo para cada registro.
+O Kubelet mesclará qualquer `imagePullSecrets` em um único `.docker/config.json` virtual
+
+## {{% heading "whatsnext" %}}
+
+* Read the [OCI Image Manifest Specification](https://github.com/opencontainers/image-spec/blob/master/manifest.md)
```
Comment (devlware) [inline content/pt/docs/concepts/containers/images.md:291]:
```suggestion
* Leia a [OCI Image Manifest Specification](https://github.com/opencontainers/image-spec/blob/master/manifest.md)
```

**Review by devlware — CHANGES_REQUESTED** [review]
Thanks for helping, I left some suggestions!

**Comment by HugoPfeffer** [comment]
Thanks for your suggestions @devlware! 

Since it's a lot of changes and I found another typo at line 277, I decided to make a new commit with all the changes. I don't know if that's recommended. If not, please tell me.

Sorry if it had too many errors. It's my first time doing a docs translations so I'm still getting the hang of it :)

**Comment by devlware** [comment]
Hi @HugoPfeffer no problem with the "many" changes, it happens to me all the time. Thanks again for putting your time on this project. :)
Regarding commit, there is no problem in doing that.

File: content/pt/docs/concepts/containers/images.md [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+
+Em vez disso, especifique uma tag significativa, como `v1.42.0`.
+{{< /caution >}}
+
+## Atualizando imagens
+
+A política padrão de pull é `IfNotPresent` a qual faz com que o
+{{<glossary_tooltip text = "kubelet" term_id = "kubelet">}} ignore 
+o processo de *pull* da imagem, caso a mesma já exista. Se você prefere sempre forçar o processo de *pull*, 
+você pode fazer como exemplo:
```
Comment (femrtnz) [inline content/pt/docs/concepts/containers/images.md:50]:
```suggestion
você pode seguir uma da opções:
```

**Comment by femrtnz** [comment]
Thanks for contributing @HugoPfeffer !!
/lgtm
/approve

**Pull request merged**